resolwe_bio/processes/workflows/geo_import.py

```diff
@@ -0,0 +1,343 @@
+            process_inputs["advanced"]["min_read_len"] = inputs.advanced.min_read_len
+        if inputs.advanced.min_read_len:
+            process_inputs["advanced"]["min_read_len"] = inputs.advanced.min_read_len
+
+        sample_info = {}
+        for name, gsm in gse.gsms.items():
+            sample_found = re.findall(
+                r"(SRX\d{6,8})", str(gse.gsms[name].relations["SRA"])
+            )
+            if sample_found:
+                for srx_id in sample_found:
+                    sample_info[srx_id] = name
+                    sra_url = f"http://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?save=efetch&db=sra&rettype=runinfo&term={srx_id}"
+                    info_file = f"{gse.name}.csv"
+                    run_info = requests.get(sra_url).content
+                    if run_info == b"\n":
+                        self.error(
+                            f"Failed to fetch SRA runs for project {srx_id} belonging to {gse.name}."
+                        )
+                    else:
+                        with open(info_file, "wb") as handle:
+                            handle.write(requests.get(sra_url).content)
+                    run_info = pd.read_csv(
+                        info_file, usecols=["Run", "SampleName", "LibraryLayout"]
+                    )
+                    run_info = run_info.set_index("Run", drop=False)
+
+                    process_inputs["sra_accession"] = run_info.index.values.tolist()
+                    assert run_info.nunique().loc["LibraryLayout"] == 1
+                    lib_type = run_info["LibraryLayout"].iloc[0]
+
+                    if lib_type == "PAIRED":
+                        self.run_process("import-sra-paired", process_inputs)
+                    elif lib_type == "SINGLE":
+                        self.run_process("import-sra-single", process_inputs)
+                    else:
+                        self.error(
+                            f"Unsupported library layout expected SINGLE or PAIRED but got {lib_type}."
+                        )
+
+                    entity_name = process_inputs["sra_accession"][0]
+                    sra_data = Data.filter(entity__name=entity_name)[-1]
+                    sra_data.entity.name = srx_id
+            else:
+                self.error(
+                    f"Matching SRX accession number for {name} was not found in GEO metadata."
+                )
+
+        return pd.DataFrame(
+            sample_info.items(), columns=["Accession", "SampleName"]
+        ).set_index("Accession", drop=False)
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        if not re.match(r"(GSE\d{1,8})", inputs.gse_accession):
+            self.error(
+                f"GEO series accessions (GSE) are supported but {inputs.gse_accession} was provided."
+            )
+
+        try:
+            gse = GEOparse.get_GEO(geo=inputs.gse_accession, destdir="./")
+        except IOError as e:
+            print(e)
```

Comment:
Is this print required or it was used just for the debugging?

Reply:
This is redundant as the error message `e` provides almost no new information.